api/src/main/java/org/apache/iceberg/types/JavaHashes.java

```diff
@@ -26,6 +26,10 @@ public class JavaHashes {
   private JavaHashes() {}
 
   public static int hashCode(CharSequence str) {
+    if (null == str) {
```

Comment:
there are places in the codebase that use `CharSequenceWrapper.wrap(...)` and then set the wrapper to `null`. The issue was never uncovered because equals/hashCode was never called on those `CharSequenceWrapper` instances.

When adding `TestCharSequenceWrapper` I noticed that this failed with a NPE, so I fixed this here and also added some tests around this